scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -197,6 +190,18 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
       })
     }
 
+  /**
+   * Apply BeamSQL query to this SCollection
+   */
+  // this method is not strictly necessary but using a invariant type instead of
+  // simple (SCollection[T] => SCollection[U]) helps with type inference
+  def sql[U](q: Query[T, U]): SCollection[U] = transform(q.query)(q)
+
+  def sqlJoin[B, U](other: SCollection[B])(q: Query2[T, B, U]): SCollection[U] =
```

Comment:
I don't like that. It feels really odd to have to use a different function just because there's a `JOIN` in your query.

Reply:
yeah agreed, I was getting some implicit divergence here ... so I took the easy route just to get this out of the door